### PR TITLE
Is1884/using integration library

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Simplifies the service project resulting from this cookiecutter by delegating tooling and standard testing to [osparc-simcore's service library](https://github.com/ITISFoundation/osparc-simcore/tree/master/packages/service-integration) or ``ooil`` in short.

We have moved most of the code in the cookiecutter to ``ooil`` library (which lives in osparc-simcore repo). The idea is that there is now **a single source of truth** : basically the ``metadata.yml`` and ``docker-compose-build.yml`` files. Then ``ooil`` can deduce everything from it. 

Next iteration will introduce the ``.osparc`` folder approach (but one step at time :-) )

This leads to a much simpler layout! Just FYI, the  **resulting service project**, compared to the previous version :
- removes ``tools`` folder code
- removes ``versioning`` folder
- removes ``VERSION`` files (can be regenerated with ``ooil`` 
- replaces test code by ``assert_*`` helper functions ( ``test`` skeleton is maintained in this iteration but could be further simplified)
- moved gitlab/github  specific routines in ``ci`` to ``.gitlab`` and ``.github`` folders resp
- improved Makefile recipes ( w/o breaking the interface in this iteration)

## How to test

Could you please try this with one of your services? To create a service project using this ``cookiecutter + python 3.9``  just
```cmd
$ cookiecutter --checkout=is1884/using-integration-library gh:pcrespov/cookiecutter-osparc-service
```
Then ``cd`` the "baked" project and
```cmd
$ ls -la
$ make devenv 
$ source .venv/bin/activate
$ ooil --version
$ make build
$ make tests

# then can try other recipes
$ make help
$ make VERSION
$ make replay
``` 

## Related issue number

- ITISFoundation/osparc-issues#694 : proactive onboarding of models and services (Y6)
- ITISFoundation/osparc-issues#304: (Part 2) Prep2Go: creating features to support complex S4L scripts
- ITISFoundation/osparc-issues#693: creation of own oSPARC services

## Checklist

- [x] change version
- [x] add reference to master
- [x] fix failing tests: issue with ``source``-ing in a ``subprocess``! (@sanderegg any idea on ``test_bake_project.py`` issue?)
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
+# SEE https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-code-owners#example-of-a-codeowners-file
+
 
 # files and folders recursively
-/                  @sanderegg
+Makefile                                  @pcrespov @sanderegg
+
+
+# NOTE: '/' denotes the root of the repository
+/.github/                                 @sanderegg @pcrespov
```
Comment (sanderegg) [inline .github/CODEOWNERS:12]: you're going to granular here.

File: .github/workflows/github-ci.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python: [3.8]
+        python: [3.9]
```
Comment (sanderegg) [inline .github/workflows/github-ci.yml:16]: buildx is also old. you can put ```v0.8.2```

File: .vscode/launch.template.json [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      "host": "127.0.0.1",
+      "pathMappings": [
+        {
+          "localRoot": "${workspaceFolder}",
+          "remoteRoot": "/devel"
+        }
+      ]
+    },
+    {
+      "name": "Python: Remote Attach datcore-adapter",
```
Comment (sanderegg) [inline .vscode/launch.template.json:40]: hmm not sure that makes sense to have this here. in this context what is the api-server?

File: .vscode/settings.template.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  "python.linting.enabled": true,
+  "[python]": {
+    "editor.detectIndentation": false,
+    "editor.tabSize": 4
+  },
+  "[makefile]": {
+    "editor.insertSpaces": false
+  },
+  "python.testing.pytestEnabled": true,
+  "shellcheck.executablePath": "${workspaceFolder}/scripts/shellcheck.bash",
```
Comment (sanderegg) [inline .vscode/settings.template.json:39]: this is not available right?

File: {{cookiecutter.project_slug}}/.gitlab/gitlab-ci.yml [hunk 5]
```diff
@@ -53,7 +53,7 @@ remove_{{ cookiecutter.project_slug }}_builds:
     - export DOCKER_PROJECT=$SC_CI_PROJECT_PATH_NAME/simcore/services/{%- if cookiecutter.project_type == "computational" -%}comp{%- elif cookiecutter.project_type == "dynamic" -%}dynamic{%- endif -%}/{{ cookiecutter.project_name.lower().replace(' ', '-') }}
     - export API_USER=$SC_CI_TESTING_REGISTRY_USER
     - export API_TOKEN=$SC_CI_TESTING_REGISTRY_PORTUS_TOKEN
-    - bash $SC_CI_{{ cookiecutter.project_package_name.upper() }}_LOCATION/ci/delete-image-from-registry-portus.bash # this will fail if registry is not available through Portus
+    - bash $SC_CI_{{ cookiecutter.project_package_name.upper() }}_LOCATION/.gitlab/delete-image-from-registry-portus.bash # this will fail if registry is not available through Portus
```
Comment (sanderegg) [inline {{cookiecutter.project_slug}}/.gitlab/gitlab-ci.yml:56]: I guess the portus stuff shall be removed... it's outdated

File: {{cookiecutter.project_slug}}/Makefile [hunk 6]
```diff
@@ -112,14 +119,19 @@ tests: tests-unit tests-integration ## runs unit and integration tests
...
-	@$(call _bumpversion,$<,version-service-)
+version-service-patch version-service-minor version-service-major: $(METADATA) ## kernel/service versioning as patch
+	ooil bump-version --metadata-file $<  --upgrade $(subst version-service-,,$@)
+	# syncing metadata upstream
+	@$(MAKE) VERSION
+	@$(MAKE) docker-compose-meta.yml
+
+.PHONY: version-integration-patch version-integration-minor version-integration-major
+version-integration-patch version-integration-minor version-integration-major: $(METADATA) ## integration versioning as patch (bug fixes not affecting API/handling), minor/major (backwards-compatible/INcompatible API changes)
+	ooil bump-version --metadata-file $<  --upgrade $(subst version-integration-,,$@) integration-version
```
Comment (sanderegg) [inline {{cookiecutter.project_slug}}/Makefile:132]: nice

**Review by sanderegg — DISMISSED** [review]
thanks a lot. I see a lot of things hat could be improved:
- non using docker buildX
- in the .gitlab folder there are a few deprecated things
- I guess same is true for .github...
anyway, very very nice.

**Review by elisabettai — COMMENTED** [review]
Looks very very promising! Looking forward to try it out (I haven't tested yet with your instructions).

**Pull request merged**